packages/@lwc/engine-core/src/framework/hooks.ts

```diff
@@ -36,6 +37,14 @@ function setElementShadowToken(elm: Element, token: string | undefined) {
     (elm as any).$shadowToken$ = token;
 }
 
+function setLightDomScopingTokenIfNecessary(elm: Element, owner: VM) {
+    const { cmpTemplate } = owner;
+    const token = cmpTemplate?.stylesheetToken;
+    if (!isUndefined(token) && hasScopedStyles(cmpTemplate)) {
```

Comment:
I don't think we can afford to invoke `hasScopedStyles` for each element in the template. It's probably something worth adding to the `owner.context`.

Reply:
Hm, maybe `setStyleTokensAndObserve`? That seems to be what it's doing, for both synthetic shadow and light DOM scoped styles.

Or I could move the light DOM style scoping to a new hook.

Reply:
Fixed it so this is only invoked once (in prod mode). In dev mode, it's also invoked a second time to check that shadow components aren't using scoped styles.